set-remove_channel.py

```diff
@@ -0,0 +1,667 @@
+import serial.tools.list_ports
+from meshtastic.serial_interface import SerialInterface
+from meshtastic.tcp_interface import TCPInterface
+from meshtastic.ble_interface import BLEInterface
+from meshtastic.util import fromPSK
+from pubsub import pub
+import time
+try:
+    from meshtastic.protobuf import channel_pb2, admin_pb2, portnums_pb2
+except ImportError:
+    from meshtastic import channel_pb2, admin_pb2, portnums_pb2
+import sys
+import argparse
+import os #OS detect
+os.system('') #required for colorama on win cmd
+from colorama import Fore #colors
+
+#Color scheme:
+#LIGHTRED_EX: errors
+#LIGHTMAGENTA_EX: warnings
+#LIGHTBLUE_EX: informational
+#GREEN: only used when script succeeds
+
+def keypress(): #input single character, cross-platform
+    if os.name == 'nt':
+        import msvcrt
+        return msvcrt.getch().decode('utf-8')
+    else:
+        import tty
+        import termios
+        fd = sys.stdin.fileno()
+        old_settings = termios.tcgetattr(fd)
+        try:
+            tty.setraw(fd)
+            char = sys.stdin.read(1)
+        finally:
+            termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
+        return char
+
+def exitscript():
+    print(f"{errormsg}{Fore.LIGHTRED_EX}Quitting...{Fore.RESET}")
+    quit()
+
+requestIds = []
+via = ""
+errormsg = ""
+gotResponse = False
+if len(sys.argv) == 1: #are there any arguments? if not, use prompts
+        print(f"""\nConnection method to local node:
+    {Fore.LIGHTBLUE_EX}1.{Fore.RESET} USB Serial
+    {Fore.LIGHTBLUE_EX}2.{Fore.RESET} Network/TCP
+    {Fore.LIGHTBLUE_EX}3.{Fore.RESET} Bluetooth/BLE
+    {Fore.LIGHTBLUE_EX}0.{Fore.RESET} Quit""")
+        i = 0
+        key = "X" #initial value for keypress detector
+        via = "" #port, IP/hostname or BLE mac address/device name
+        while key not in ("1", "2", "3", "0"):
+            key = keypress()
+            i += 1
+            match key:
+                case "1":
+                    method = "usb"
+                    readablemethod="USB Serial"
+                    availableports = [comport.device for comport in serial.tools.list_ports.comports()] #mian - this does not work on linux. Leaving this one to you. My head hurts.
```

Comment:
I can probably get this working. There's a `meshtastic.util.findPorts` function that's what the CLI uses for this that I can probably just shove in here somehow, haha.